eZ/Publish/Core/FieldType/Image/IO/Legacy.php

```diff
@@ -169,6 +184,10 @@ public function loadBinaryFileByUri($binaryFileUri)
 
     public function getFileContents(BinaryFile $binaryFile)
     {
+        if ($this->draftIOService->exists($binaryFile->id)) {
```

Comment:
Not really, just about what was conceptually convenient at the time. As I recall, the class was divided on this before, too. (Look further up, and it's draft first again.) On line 144 I replaced a non-draft only call with a try-catch that does both, guess it just felt logical to keep the non-draft first there.

Reply:
Shouldn't all cases check non draft first? Otherwise  there occurs a overhead of looking up draft first also for main use case: published content

Reply:
At least in this method I think the overhead is the same in both cases: Check if one file exists, if yes load it, if no load another file. So there is always one exists call and one load.

Reply:
We could switch around the two sections at the top of loadBinaryFile(), that would help a bit. https://github.com/ezsystems/ezpublish-kernel/pull/2203/files/cfc3d5e3fea03880e3b156573a07977ed85dcb94#diff-fb574f077a9d9847031cd4f0a93a11aaL134

Reply:
I tend to agree with Andre here: most common case is file will exist. If we check for draft file first, we end up with 2 exists calls instead of one.
Otoh I am not 100% sure about the case of a draft for v2 or later: is the filename always unique (ie. does it include the version nr)?. Otherwise there is the risk of loading current file (v X) instead of draft (v X+1)

Reply:
Not in this method, there is only one exists call here, so which we check first doesn't matter. Same number of calls either way. But I see the point for some of the other methods, and if we reorder those, it makes sense to reorder this one too, just for the sake of readability.

As I recall, versioned paths include `-versioned`, which the published one doesn't have. Otherwise, the `isDraftImagePath()` / `isPublishedImagePath()` methods wouldn't make sense.